src/mp_integer.hpp

```diff
@@ -2294,6 +2294,21 @@ class mp_integer
 		{
 			return mpz_view(*this);
 		}
+		/// Get an \p non-const mpz_ptr to dynamic storage of \p this.
+		/**
+		 * This method will return a pointer to the dynamic storage, mpz_struct_t. If \p this is currently stored in static
+		 * storage, \p this will be promoted to dynamic storage first
+		 *
+		 * Note that \p this will remain stored in dynamic storage even if after operations on \p this, value of \p this can
+		 * be stored in static storage.
+		 *
+		 * @return an \p non-const mpz_ptr to dynamic storage of \p this.
+		 */
+		mpz_ptr get_mpz_ptr()
+		{
+			if (is_static()) promote();
```

Comment:
As a matter of style, I prefer to put the brackets on an ``if`` statement even if it fits in a single line.